### PR TITLE
By default, don’t pass a number format to vega.

### PR DESCRIPTION
**Pull request by domoritz**
With this change, by default we don't see the legend label issues from #497.

### CODE REVIEW

File: test/compile/Model.test.ts [hunk 1]
```diff
@@ -60,9 +60,12 @@ describe('Model', function() {
         mark: "point",
         encoding: {
           x: {field:'a', type: "quantitative"}
+        },
+        config: {
+          numberFormat: 'foo'
```
Comment (kanitw) [inline test/compile/Model.test.ts]: Although `’foo’` is fine placing here for a purpose of unit testing, I prefer to a value that is a valid in real life setting as well.

**Comment by kanitw** [comment]
Another general question that I should have noted in #497 is whether we should keep the `config.numberFormat` and `config.timeFormat`?       

If yes, we should remove `FIXME` comment in `config.schema.ts`.

**Comment by domoritz** [comment]
I think we will always need a number format. Even if we add different number formats for special cases, we still need a config option.

**Comment by domoritz** [comment]
Done

**Pull request merged**